syntax = "proto3";

package streaming;

service StreamingService {
    rpc GetStream (StreamRequest) returns (StreamResponse) {}
    rpc HealthCheck (HealthRequest) returns (HealthResponse) {}
}

message StreamRequest {
    string video_id = 1;
}

message StreamResponse {
    string stream_url = 1;
    string edge_node_id = 2;
}

message HealthRequest {}

message HealthResponse {
    bool status = 1;
}